### PR TITLE
makes EasyPostObject resolution code more DRY

### PR DESCRIPTION
**Pull request by jontsai**
How we determine what object to deserialize an API request's JSON response into is quite messy. This PR improves the logic regarding that, and makes the code more DRY and maintainable.

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+force_grid_wrap=True
+multi_line_output=3
+known_first_party=
+known_third_party=
+sections=FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
+import_heading_future=
+import_heading_stdlib=
+import_heading_thirdparty=
+import_heading_firstparty=
+import_heading_localfolder=
```
Comment (jchen293) [inline .isort.cfg:13]: Can we remove those empty assignments?

**Review by jchen293 — APPROVED** [review]

File: easypost/easypost_object.py [hunk 2]
```diff
@@ -1,86 +1,75 @@
...
+    "shp": "Shipment",
+    "shpinvrep": "Report",
+    "shprep": "Report",
+    "trk": "Tracker",
+    "trkrep": "Report",
+    "user": "User",
+}
+
+
+OBJECT_CLASS_NAME_OVERRIDES = {
```
Comment (Justintime50) [inline easypost/easypost_object.py:41]: Not all report types are represented here.

**Review by Justintime50 — COMMENTED** [review]
Can we update the PR summary with the goal and outcome of this are?

File: .isort.cfg (hunk 1 above)
Comment (nwithan8) [reply .isort.cfg]: Resolved in 461bee55ae4af25242c74b87dfa5b0360cc188e8

File: easypost/easypost_object.py (hunk 2 above)
Comment (nwithan8) [reply easypost/easypost_object.py:40]: Resolved in 461bee55ae4af25242c74b87dfa5b0360cc188e8

File: easypost/easypost_object.py [hunk 3]
```diff
@@ -1,86 +1,75 @@
...
+    "pl": "PostageLabel",
+    "plrep": "Report",
+    "prcl": "Parcel",
+    "rate": "Rate",
+    "refrep": "Report",
+    "rfnd": "Refund",
+    "sf": "ScanForm",
+    "shp": "Shipment",
+    "shpinvrep": "Report",
+    "shprep": "Report",
```
Comment (Justintime50) [inline easypost/easypost_object.py:34]: We're missing other report types here

File: easypost/easypost_object.py [hunk 4]
```diff
@@ -1,86 +1,75 @@
...
+    "trk": "Tracker",
+    "trkrep": "Report",
+    "user": "User",
```
Comment (Justintime50) [inline easypost/easypost_object.py:37]: `webhook` is missing from the list. We may want to check it to ensure nothing else is missing.

**Review by Justintime50 — DISMISSED** [review]

File: easypost/easypost_object.py (hunk 4 above)
Comment (nwithan8) [reply easypost/easypost_object.py:37]: It's "hook"

Comment (Justintime50) [reply easypost/easypost_object.py:37]: Ah thanks, skipped right over it ha.

**Review by jchen293 — APPROVED** [review]

**Pull request merged**